### DISCUSSION TITLE
Updating values in-place

### DISCUSSION THREAD

**Discussion by tksharpless** [nlohmann/json#2864] [Q&A] [answered]
I need to create copies of large Json files, with some values changed but all keys preserved in the original order.
For the second requirement, ordered_json works nicely, but I'm having trouble implementing the first.
I would prefer to update the values in the json object originally parsed from the input file, as other parts of the application treat that as the current value. 
I tried  json::update() but it is clear that this can't work, as it destroys everything not mentioned in the update object.
With json_pointer I should be able to target a single value, but it needs to be templated with the value type; and I' had trouble writing compilable code using json_pointer.  So a generic update() function seems difficult.

So instead I am using explicit references, in code like 
> auto json &  arr = top["boop"]["foo"];  // array  reference
> arr = some_json_array;
or
> auto json &  obj = top["boop"]["foo"][3]["baz"];  // object reference
> obj = some_json_object;
As expected, these update top in-place.  But there is a problem with null values
In the  first form arr = obj.array() gives an empty array represented by "foo": [] in the output. Fine.
In the second, obj = obj.object() sets the stored value of baz to null; however the serialized output is 
>"baz": {}
which is not accepted by the external app, it wants to see 
>"baz": null
Is there any way I can make that happen within nlohmann_json?

**Comment by nlohmann** [comment, marked as answer]
You can use the following syntax to store a `null` value:

```cpp
json foo = "string";

// using the default constructor
foo = json();

// or using the nullptr literal
foo = nullptr;
```

**Reply by tksharpless** [reply]
Thanks.  That is very helpful.
